### PR TITLE
use placement api for assinging virt-handler pod

### PR DESCRIPTION
**Pull request by ashleyschuett**
**What this PR does / why we need it**:

We should not be manually changing the virt-handler daemonset, as once we introduce reconcile for this resource the test will break. Instead this PR changes to use the workload placement api to change the location of the virt-handler pods through the KubeVirt CR. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/vmi_lifecycle_test.go [hunk 1]
```diff
@@ -571,47 +571,68 @@ var _ = Describe("[rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:compon
 			})
 		})
 
-		Context("[Serial]when virt-handler is not responsive", func() {
+		FContext("[Serial]when virt-handler is not responsive", func() {
```
Comment (davidvossel) [inline tests/vmi_lifecycle_test.go:574]: remove the focus here

File: tests/vmi_lifecycle_test.go [hunk 2]
```diff
@@ -645,15 +666,17 @@ var _ = Describe("[rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:compon
...
 				// Restore virt-handler daemonset
-				ds, err := virtClient.AppsV1().DaemonSets(virtHandler.Namespace).Get("virt-handler", metav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred(), "Should get virthandler successfully")
-				ds.Spec.Template.Spec.Affinity = nil
-				_, err = virtClient.AppsV1().DaemonSets(virtHandler.Namespace).Update(ds)
-				Expect(err).ToNot(HaveOccurred(), "Should update virthandler successfully")
+				kv := tests.GetCurrentKv(virtClient)
+				kv.Spec.Workloads = dsOriginalConfiguration
+				_, err = virtClient.KubeVirt(virtHandler.Namespace).Update(kv)
+				Expect(err).ToNot(HaveOccurred(), "Should update kubevirt infra placement")
```
Comment (davidvossel) [inline tests/vmi_lifecycle_test.go:679]: can you get away with calling tests.RestoreKubeVirtResource() here?

File: tests/vmi_lifecycle_test.go [hunk 3]
```diff
@@ -571,47 +571,68 @@ var _ = Describe("[rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:compon
...
-							NodeSelectorTerms: []k8sv1.NodeSelectorTerm{
-								{MatchExpressions: []k8sv1.NodeSelectorRequirement{
-									{Key: "kubernetes.io/hostname", Operator: "NotIn", Values: []string{nodeName}},
-								}},
-							},
-						},
-					},
-				}
+
 				_, err = virtClient.AppsV1().DaemonSets(virtHandler.Namespace).Update(ds)
```
Comment (davidvossel) [inline tests/vmi_lifecycle_test.go:627]: this can be removed now, right?

**Review by davidvossel — COMMENTED** [review]
The test makes sense. I had a few minor comments

**Comment by ashleyschuett** [comment]
@davidvossel thank you for the review, I have updated it!

**Review by davidvossel — APPROVED** [review]
/lgtm
/approve

**Pull request merged**